tls/s2n_client_hello.h

```diff
@@ -36,6 +36,20 @@ struct s2n_client_hello {
     unsigned int callback_invoked:1;
     unsigned int callback_async_blocked:1;
     unsigned int callback_async_done:1;
+    /*
+     * Marks if the client hello has been parsed.
+     *
+     * While a client_hello is only parsed once, it is possible to parse
+     * two different client_hello during a single handshake if the server
+     * issues a hello retry.
+     */
+    unsigned int parsed:1;
```

Comment:
Can you check the size of raw_message to check if the client hello was parsed?

Reply:
Is there a reason why the current implementation is bad?

Not sure, but that is also more complicated to reason about and can potentially break if something changes about how we parse client hello. We saw this in the client_hello gets parsed 2 times when we send a hello_retry.

Reply:
It can might be simpler to track state in one place instead of two. I'm fairly sure that whether or not the client hello / raw_message exists is the "source of truth" for whether or not the client hello has been parsed.

Your new "parsed" flag handles retries because you reset "parsed" on a retry, but we already reset "raw_message" on a retry: https://github.com/aws/s2n-tls/blob/b081a909caccf71eedd5d71526c489c2a0417983/tls/s2n_client_hello.c#L163-L175

Reply:
spoke offline: ill try checking the stuffer and see if it works

Reply:
addressed in https://github.com/aws/s2n-tls/pull/3230/commits/2858e16f2c47dc25b6cbf1fdabca30440ac8704b